tests/calico_cni_k8s_test.go

```diff
@@ -248,6 +282,10 @@ var _ = Describe("Kubernetes CNI tests", func() {
 					Type:      syscall.RTN_UNICAST,
 				})))
 
+			// Delete pod
+			ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name)
```

Comment:
Why do we need this change? I'm not sure I'm following this.

Reply:
I think we need the Pod to exist at this point - `testutils.DeleteContainer` is calling the CNI plugin, which expects the pod to exist in the API.

Reply:
I think I tried to delete pod here because we are checking WEP count later. I have made changes that we only check WEP count if it is etcd datastore and added 
`defer ensurePodDeleted(clientset, testutils.K8S_TEST_NS, name)`  to make sure we cleanup pod.

Reply:
This case isn't doing a defer though, so I don't think it's quite right. I think so long as we do this at the end it is OK but we should make sure they're all properly deferred or, ideally, in an `AfterEach`.

Reply:
Yes. Ideally all defer should be in `AfterEach`. However, there are cases that two pod with different names been created where we use `defer`